g11n-ws/modules/md-service-i18n-l3/src/main/java/com/vmware/vip/core/messages/service/multcomponent/MultComponentService.java

```diff
@@ -110,5 +115,95 @@ private TranslationDTO getTranslation(TranslationDTO translationDTO)
 		translationDTO.setBundles(ja);
 		return translationDTO;
 	}
+	
+	
+	@SuppressWarnings("unchecked")
+    private TranslationDTO getV2Translation(TranslationDTO translationDTO)  throws DataException{
+	    List<String> locales = translationDTO.getLocales();
+        List<String> components = translationDTO.getComponents();
+        if (components == null || locales == null) {
+            throw new DataException("No component or locale");
+        }
+        
+        List<JSONObject> jsonObjectList = new ArrayList<JSONObject>();
+        JSONArray ja = new JSONArray();
+        for (String component : components) {
+            for (String locale : locales) {
+    
+                try {
+                    String json = oneComponentDao.get2JsonStr(translationDTO.getProductName(), translationDTO.getVersion(),component, locale);
+                    if (StringUtils.isEmpty(json)) {
+                        jsonObjectList.add(getNUllBundle(component, locale));
+                        continue;
+                    }
+                    JSONObject jo = (JSONObject) new JSONParser().parse(json);
+                    ja.add(jo);
+                }catch(Exception e) {
+                    jsonObjectList.add(getNUllBundle(component, locale));
+                    continue;
+                }
+            }
+        }
+        
+        
+        
+ 
+        if(ja.size()<1) {
```

Comment:
Add one case, if ja.size doesn't equal to locales.size multiply components.size.